### PR TITLE
Upgraded core to 0.94.3

### PR DESCRIPTION
**Pull request by cmelchior**
This upgrades Core to 0.94.2

It will potentially be 0.94.3 before release.

### CODE REVIEW

**Comment by dalinaum** [comment]
:+1:

**Comment by kneth** [comment]
:+1:

**Comment by dalinaum** [comment]
We can upgrade Core to 0.94.3
https://github.com/realm/realm-core/releases/tag/v0.94.3

**Comment by cmelchior** [comment]
Bump @realm/java 
Upgraded to 0.94.3 + revised changelog

File: changelog.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@
  * Added Realm.distinct() method.
  * Added RealmQuery.isValid(), RealmResults.isValid() and RealmList.isValid(). Each method checks whether the instance is still valid to use or not(for example, the Realm has been closed or any parent object has been removed).
  * Added Realm.isInTransaction() method.
+ * Updated Realm Core to version 0.94.3
```
Comment (kneth) [inline changelog.txt]: "Realm Core" -> "Realm's storage engine" (multiple places)

**Comment by dalinaum** [comment]
:+1:

**Comment by dalinaum** [comment]
We can easily change `Realm Cor` later. So I'll merge this.

**Pull request merged**